src/skmatter/_selection.py

```diff
@@ -209,8 +209,7 @@ def fit(self, X, y=None, warm_start=False):
         params = dict(ensure_min_samples=2, ensure_min_features=2, dtype=FLOAT_DTYPES)
 
         if hasattr(self, "mixing") or y is not None:
-            X, y = self._validate_data(X, y, **params)
-            X, y = validate_data(self, X, y, multi_output=True)
+            X, y = validate_data(self, X, y, multi_output=True, **params)
```

Comment:
Do we maybe also use `.ravel()` here or `multi_output=True` in the other `validate_data` calls?